subsys/net/lib/mqtt/mqtt_transport_websocket.c

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ * Copyright (c) 2019 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/** @file mqtt_transport_websocket.c
+ *
+ * @brief Internal functions to handle transport over Websocket.
+ */
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(net_mqtt_websocket, CONFIG_MQTT_LOG_LEVEL);
+
+#include <errno.h>
+#include <net/socket.h>
+#include <net/mqtt.h>
+#include <net/websocket.h>
+
+#include "mqtt_os.h"
+#include "mqtt_transport.h"
+
+/**@brief Handles connect request for Websocket transport.
```

Comment:
I have a feeling that we keep duplicating these docs needlessly. This functionality is documented in `mqtt_transport.h`, so it's probably not needed to repat it in specific transport implementations. Personally, I'd remove the docs from other transports as well.

Reply:
Agreed, I will add a patch to this PR that does that.